worker-sys/src/types/r2/checksums.rs

```diff
@@ -0,0 +1,54 @@
+use js_sys::{ArrayBuffer, Object, Reflect, Uint8Array};
+use wasm_bindgen::JsCast;
+
+#[derive(Debug, Clone)]
+pub struct R2Checksums {
```

Comment:
I'm wondering if this should be implemented as distinct objects which implement a trait, to enforce that only one can be set at a time.

Reply:
That was my initial idea, because its stated that you can only set a single hash for PUT, but put internally doesn't need to use the R2Checksums, as the hashes are set using methods.

But for GET things were different.

In practice, in my tests I found out that on GET you always had the md5 hash, plus the one you added.

And the documentation states the values as optional ArrayBuffers in a R2Checksums object, so I tried to follow that instead of creating my own api.

https://developers.cloudflare.com/r2/api/workers/workers-api-reference/#checksums

One possibility for R2Checksums was to use the same strategy as for PUT, to have an md5 hash, plus a string identifying the optional hash and the optional hash, using methods instead of fields, but it wasn't the same api and it less clear.

I thought about using a trait, specially for PUT, but just for internal use, I wasn't sure if the extra code complexity was worth it, and I intended to respect the current API as much as possible.

But we can do that no problem, just point me in the right direction.

Reply:
I also thought about using a struct for PUT internally.

```
pub enum R2ChecksumType {
    MD5,
    SHA1,
    SHA256,
    SHA384,
    SHA512,
}

struct R2Checksum {
    checksum_type: ChecksumType,
    checksum: Vec<u8>,
}
```

The trait would be something like

```
pub trait R2Checksum {
  fn algorithm(&self) -> &str; // or R2ChecksumType
  fn checksum(&self) -> Vec<u8>;
}
```

But I wasn't sure the added complexity was worth it, so I tried to minimize impact.

Although I do either option in my own code depending on the context, I organize my code differently, so that the added complexity is acceptable.